src/app.js

```diff
@@ -1 +1,44 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-console */
 'use strict';
+
+const fs = require('fs');
+
+const copyFile = (source, destination) => {
+  if (!source || !destination) {
+    throw new Error('Source and destination must be provided');
+  }
+
+  if (source === destination) {
+    throw new Error('Source and destination paths are the same');
+  }
+
+  if (!fs.existsSync(source)) {
+    throw new Error(`${source} doesn't exist`);
+  }
+
+  const sourceStats = fs.statSync(source);
+
+  if (sourceStats.isDirectory()) {
+    throw new Error('Source is a directory, not a file');
+  }
+
+  if (fs.existsSync(destination)) {
+    const destStats = fs.statSync(destination);
+
+    if (destStats.isDirectory()) {
+      throw new Error('Destination is a directory, not a file');
```

Comment:
Consider clarifying in the error message that the destination should be a file path, not a directory.